src/acl/Checklist.cc

```diff
@@ -370,9 +370,9 @@ ACLChecklist::fastCheck()
 void
 ACLChecklist::calcImplicitAnswer()
 {
-    const allow_t lastAction = (accessList && cbdataReferenceValid(accessList)) ?
-                               accessList->lastAction() : allow_t(ACCESS_DUNNO);
-    allow_t implicitRuleAnswer = ACCESS_DUNNO;
+    const Acl::Answer lastAction = (accessList && cbdataReferenceValid(accessList)) ?
```

Comment:
Please take advantage of the change to use C++11 "auto" instead of an explicit type name for lastAction.
```suggestion
    const auto lastAction = (accessList && cbdataReferenceValid(accessList)) ?
```

There are quite a few other places which used to have explicit allow_t which can also become auto now instead of Acl::Answer. The rule of thumb to use is that if the RHS of an assignment is a function, method or variable with the same type as the new variable on the LHS then you can use auto instead of repeating its base type name.

The tricky part here is that the ACCESS_* things have a different type so only use auto there if they are assigned using the type conversion wrapper Acl::Answer(ACCESS_FOO).

Reply:
> The rule of thumb to use is that if the RHS of an assignment is a function, method or variable with the same type as the new...

The rule of thumb is actually a lot simpler: If you can use `auto`, use `auto`. Same for `const`. These rules are applied to new and changed-for-other-reasons code. As is typical for rules of thumb, they come with exceptions, but they usually work fine.